src/jenkins_job_insight/main.py

```
@@ -3921,6 +3946,41 @@ async def get_ai_configs_endpoint() -> list[dict]:
+        if provider:
+            provider = provider.lower().strip()
+            models = await model_cache.list_models(provider)
+            return {"provider": provider, "models": models}
+
+        # No provider specified — return models for all known providers
+        all_models: dict[str, list[dict]] = {}
+        for p in sorted(VALID_AI_PROVIDERS):
+            try:
+                models = await model_cache.list_models(p)
+                all_models[p] = models
+            except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)